front/src/common/RollingStockSelector/RollingStockModal.tsx

```diff
@@ -88,55 +88,62 @@ function RollingStockModal({ ref2scroll }: RollingStockModal) {
     elec: false,
     thermal: false,
   });
-  const [isLoading, setIsLoading] = useState(false);
+  const [rollingStocks, setRollingStocks] = useState<LightRollingStock[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
   const [openRollingStockCardId, setOpenRollingStockCardId] = useState(rollingStockID);
   const { closeModal } = useContext(ModalContext);
 
   if (darkmode) {
     import('./RollingStockDarkMode.scss');
   }
 
-  const { rollingStocks, isSuccess, isError, error } = osrdEditoastApi.useGetLightRollingStockQuery(
-    {
-      pageSize: 1000,
-    },
-    {
-      selectFromResult: (response) => ({
-        ...response,
-        rollingStocks: sortBy(response.data?.results, ['metadata.reference', 'name']) || [],
-      }),
-    }
-  );
-  const [filteredRollingStockList, setFilteredRollingStockList] = useState<LightRollingStock[]>(
-    () => filterRollingStocks(rollingStocks, filters)
+  const { apiRollingStocks, isError, error, requestIsLoading } =
+    osrdEditoastApi.useGetLightRollingStockQuery(
+      {
+        pageSize: 1000,
+      },
+      {
+        selectFromResult: (response) => ({
+          ...response,
+          requestIsLoading: !response.data,
+          apiRollingStocks: sortBy(response.data?.results, ['metadata.reference', 'name']) || [],
```

Comment:
https://github.com/DGEXSolutions/osrd/pull/4333
I have made this PR that adds a transformResponse before the result it stored in redux.
I still have a few exceeded maximum depth, but I think it should solve your issue with the useEffect references